kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ResourceCompare.java

```diff
@@ -43,4 +52,11 @@
         result.remove(METADATA);
         return result;
     }
+
+    private static Map<String, Object> fetchLabels(Map<String, Object> map){
+        if (!map.containsKey(METADATA) || !((Map<Object, Object>)map.get(METADATA)).containsKey(LABELS)){
+            return MapUtils.EMPTY_MAP;
```

Comment:
Why not just `Collections.emptyMap()`? This adds a hard dependency on Commons Collections, which is useless (perfectly acceptable alternative directly in the JDK) and actually causes `java.lang.NoClassDefFoundError: org/apache/commons/collections/MapUtils` at runtime.

Reply:
Oups, good catch, thanks.

Reply:
np, 4.2.2 is released with the fix